packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/extension/plugin/JavaSettings.java

```diff
@@ -408,17 +408,22 @@ private void updateFlavorFactories() {
      * @return Whether to generate with Azure V1.
      */
     public boolean isAzureV1() {
-        return isAzureV1(this.flavor);
-    }
-
-    private static boolean isAzureV1(String flavor) {
         return "azure".equalsIgnoreCase(flavor);
     }
 
+    /**
+     * Whether to generate with Azure V2.
+     *
+     * @return Whether to generate with Azure V2.
+     */
     public boolean isAzureV2() {
         return "azurev2".equalsIgnoreCase(this.flavor);
     }
 
+    public boolean isUnbranded() {
+        return !isAzureV1() && !isAzureV2();
+    }
```

Comment:
Not related to this PR. `isVanilla` check may need update as well..
https://github.com/microsoft/typespec/blob/2c2aef189281d8d3c0849cad7a06c1aea1c087b6/packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/extension/plugin/JavaSettings.java#L1173-L1175

I'll do it in a follow-up PR.

Reply:
Seems still OK (assume v2 does not have vanilla).

Here it is limited to v1.